webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/TechniqueCompilationCache.scala

```diff
@@ -0,0 +1,167 @@
+case class CompilationStatusErrors(techniquesInError: NonEmptyChunk[EditorTechniqueError]) extends CompilationStatus
+
+/**
+  * Update and get latest global techniques compilation status
+  */
+trait ReadTechniqueCompilationStatusService {
+
+  def get(): IOResult[CompilationStatus]
+
+}
+
+/**
+  * Update technique compilation status from technique compilation output and technique info
+  */
+trait WriteTechniqueCompilationStatusService {
+
+  def update(technique: EditorTechnique, output: TechniqueCompilationOutput): UIO[CompilationStatus]
+
+}
+
+/**
+  * Service to update global technique compilation status and get the latest one from filesystem
+  */
+class TechniqueCompilationStatusService(
+    readTechniques:           EditorTechniqueReader,
+    techniqueCompiler:        TechniqueCompiler,
+    compilationStatusService: WriteTechniqueCompilationStatusService
+) extends ReadTechniqueCompilationStatusService {
+
+  override def get(): IOResult[CompilationStatus] = {
+    readTechniques.readTechniquesMetadataFile.flatMap {
+      // _._3 errors are not compilation errors but error on techniques, we ignore them for status
+      case (techniques, _, _) => {
+
+        techniques match {
+          case Nil        =>
+            ConfigurationStatusLoggerPure
+              .trace(
+                s"Get compilation status : no technique found"
+              )
+              .as(CompilationStatusAllSuccess)
+          case c @ _ :: _ =>
+            // aggregate all outputs,
+            // take last status,
+            // and if no output at all it means all is success
+
+            val outputs: IOResult[NonEmptyChunk[Option[CompilationStatus]]] = ZIO
+              .foreach(NonEmptyChunk.fromCons(c))(technique => {
+                techniqueCompiler
+                  .getCompilationOutput(technique)
+                  .flatMap(ZIO.foreach(_)(compilationStatusService.update(technique, _)))
+              })
+            outputs.map(_.reverse.collectFirst { case Some(out) => out }
+              .getOrElse(CompilationStatusAllSuccess)) <* ConfigurationStatusLoggerPure.trace(
+              s"Get compilation status : read ${techniques.size} editor techniques to update compilation status with"
+            )
+        }
+      }
+    }
+  }
+
+}
+
+/**
+  * Technique compilation output needs to be saved in a cache (frequent reads, we don't want to get files on the FS every time).
+  * 
+  * This cache is a simple in-memory one which only saves errors, 
+  * so it saves only compilation output stderr message in case the compilation failed.
+  * It notifies the lift actor on update of the compilation status.
+  * 
+  * It is used mainly to get errors in the Rudder UI, and is updated when API requests to update techniques are made,
+  * when technique library is reloaded
+  */
+class TechniqueCompilationErrorsCache(
+    actor: SimpleActor[UpdateCompilationStatus],
+    ref:   Ref[Map[(BundleName, Version), EditorTechniqueError]]
```

Comment:
for properties part of an API like to constructor attribute, try to name things for what they model, not what they are: here, it's the `errorBase` (even if it's implemented with a ref)

Reply:
:white_check_mark: https://github.com/Normation/rudder/pull/5905/commits/ed28eef83afc9fb4aef84d7d020a68ce36b4d63f